File: libxlsxwriter-ObjC/libxlsxwriter-ObjC/ExamplePickerTableViewController.m
//
//  ExamplePickerTableViewController.m
//  libxlsxwriter-ObjC
//
//  Created by Ludovico Rossi on 11/11/15.
//  Copyright © 2015 Ludovico Rossi. All rights reserved.
//

#import "ExamplePickerTableViewController.h"
#import "ExampleManager.h"

static NSString * const kExampleCellIdentifier = @"ExampleCell";


@interface ExamplePickerTableViewController ()

@property (readonly, nonatomic) ExampleManager *exampleManager;

@end


@implementation ExamplePickerTableViewController
@synthesize exampleManager=_exampleManager;

#pragma mark - Table View Data Source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.exampleManager.numberOfExamples;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:kExampleCellIdentifier];
    cell.textLabel.text = [self.exampleManager titleOfExampleAtIndex:indexPath.row];
    cell.detailTextLabel.text = [self.exampleManager descriptionOfExampleAtIndex:indexPath.row];

    return cell;
}

#pragma mark - Example Manager

- (ExampleManager *)exampleManager {
    if (!_exampleManager) {
        _exampleManager = [[ExampleManager alloc] init];
    }
    
    return _exampleManager;
}

@end

File: libxlsxwriter-Swift/libxlsxwriter-Swift/ExamplePickerTableViewController.swift
//
//  ExamplePickerTableViewController.swift
//  libxlsxwriter-Swift
//
//  Created by Ludovico Rossi on 12/11/15.
//  Copyright © 2015 Ludovico Rossi. All rights reserved.
//

import UIKit

class ExamplePickerTableViewController: UITableViewController {
    private static let kExampleCellIdentifier = "ExampleCell"
    private static let kViewExampleSegueIdentifier = "ViewExample"
    
    private lazy var examples = DefaultExamples.createExamples() as! [Example]
    
    // MARK: - Table View Data Source
    
    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return examples.count
    }
    
    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier(ExamplePickerTableViewController.kExampleCellIdentifier)!
        let example = examples[indexPath.row]
        cell.textLabel!.text = example.title
        cell.detailTextLabel!.text = example.subtitle
        
        return cell
    }
    
    // MARK: - View Online
    
    @IBAction func openWebPage() {
        let pageURL = NSURL(string: "http://libxlsxwriter.github.io/examples.html")!
        UIApplication.sharedApplication().openURL(pageURL)
    }
    
    // MARK: - Segues
    
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        if segue.identifier == ExamplePickerTableViewController.kViewExampleSegueIdentifier {
            let vc = segue.destinationViewController as! ExampleViewerViewController
            vc.example = examples[tableView.indexPathForSelectedRow!.row]
        }
    }
}
